### PR TITLE
[15.0][MIG] product_pack_category_restriction

### PR DESCRIPTION
**Pull request by JasminSForgeFlow**
Standard Migration

@ForgeFlow

### CODE REVIEW

File: product_pack_category_restriction/models/product_product.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+from odoo import api, models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    @api.onchange("categ_id")
+    def _onchange_categ_id(self):
+        res = super()._onchange_categ_id()
```
Comment (GuillemCForgeFlow) [inline product_pack_category_restriction/models/product_product.py]:
```suggestion
```
There is no _onchange_categ_id defined before.

File: product_pack_category_restriction/models/product_template.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    @api.constrains("pack_ok")
+    def _check_pack_ok(self):
+        products = self.filtered(lambda p: p.pack_ok)
+        if products and any(not p.categ_id.pack_ok for p in products):
+            raise ValidationError(_("Pack Products must be part of a Pack Category."))
+
+    @api.onchange("categ_id")
+    def _onchange_categ_id(self):
+        res = super()._onchange_categ_id()
```
Comment (GuillemCForgeFlow) [inline product_pack_category_restriction/models/product_template.py]:
```suggestion
```
Same here

**Review by GuillemCForgeFlow — CHANGES_REQUESTED** [review]

File: product_pack_category_restriction/models/product_product.py (hunk 1 above)
Comment (JasminSForgeFlow) [reply product_pack_category_restriction/models/product_product.py]: Updated,
Thanks

File: product_pack_category_restriction/models/product_template.py (hunk 2 above)
Comment (JasminSForgeFlow) [reply product_pack_category_restriction/models/product_template.py]: Updated,
Thanks

**Review by GuillemCForgeFlow — APPROVED** [review]
LGTM! 👍🏽

File: product_pack_category_restriction/tests/test_product_pack_product_category.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# Copyright 2023 ForgeFlow S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.exceptions import ValidationError
+from odoo.tests import TransactionCase
+
+
+class TestProductPackCategoryRestriction(TransactionCase):
+    def setUp(self):
+        super().setUp()
```
Comment (OriolMForgeFlow) [inline product_pack_category_restriction/tests/test_product_pack_product_category.py]: Change to setUpClass according to Migration Rules v15

image

Comment (JasminSForgeFlow) [reply product_pack_category_restriction/tests/test_product_pack_product_category.py]: its updated,
Thanks

**Review by OriolMForgeFlow — APPROVED** [review]
LGTM!

**Pull request closed without merging**